### ISSUE TITLE
Error loading libpq.5.dylib from Ruby pg gem in MacOS 20220829.1

### ISSUE THREAD

**Issue by filbranden** [actions/runner-images#6173]
### Description

I started getting this error in my rspec run on Mac:

```
rake aborted!
LoadError: dlopen(/Users/runner/work/vimgolf/vimgolf/vendor/bundle/ruby/2.7.0/gems/pg-1.2.3/lib/pg_ext.bundle, 9): Library not loaded: /usr/local/opt/postgresql/lib/libpq.5.dylib
  Referenced from: /Users/runner/work/vimgolf/vimgolf/vendor/bundle/ruby/2.7.0/gems/pg-1.2.3/lib/pg_ext.bundle
  Reason: image not found - /Users/runner/work/vimgolf/vimgolf/vendor/bundle/ruby/2.7.0/gems/pg-1.2.3/lib/pg_ext.bundle
```

https://github.com/igrigorik/vimgolf/runs/8119938671?check_suite_focus=true

Looking at the [release notes for image 20220829.1](https://github.com/actions/runner-images/blob/macOS-11/20220829.1/images/macos/macos-11-Readme.md) and comparing it to the previous image, I see the list of shipped utilities changed from "PostgreSQL 14.5" to "PostgreSQL 14.5 (Homebrew)", so I imagine that might have been related to this specific issue?

### Platforms affected

- [ ] Azure DevOps
- [x] GitHub Actions

### Virtual environments affected

- [ ] Ubuntu 18.04
- [ ] Ubuntu 20.04
- [ ] Ubuntu 22.04
- [ ] macOS 10.15
- [X] macOS 11
- [ ] macOS 12
- [ ] Windows Server 2019
- [ ] Windows Server 2022

### Image version and build link

20220829.1
https://github.com/igrigorik/vimgolf/runs/8119938671?check_suite_focus=true

### Is it regression?

Yes, 20220822.4, https://github.com/igrigorik/vimgolf/runs/8066604680?check_suite_focus=true

### Expected behavior

Not fail due to loading dynamic library for Postgres.

### Actual behavior

Fails due to loading dynamic library for Postgres.

### Repro steps

bundle install with pg gem and then run some Ruby code that uses it.

**Comment by miketimofeev** [comment]
Hi @filbranden!
The output in the readme is not related to the problem. The root cause is the changes in Homebrew postgres formula — it was bumped from 14 to 15 but as we still use version 14 it's now installed under the `/usr/local/opt/postgresql@14` instead of `/usr/local/opt/postgresql`
Could you try symlinking the directory` ln -s "/usr/local/opt/postgresql@14" "/usr/local/opt/postgresql"` and see if it helps?

**Comment by filbranden** [comment]
>  Could you try symlinking the directory ln -s "/usr/local/opt/postgresql@14" "/usr/local/opt/postgresql" and see if it helps?

Since this is a regression on the image itself, shouldn't you consider shipping this symlink as part of the image, for backwards compatibility?

I'm a little bit reluctant to update my action to add a symlink to an image that should just work...

Or did you mean for me to try to add the symlink to confirm it solves the issue, so it can be incorporated on the upstream image?

**Comment by miketimofeev** [comment]
@filbranden correct, if it helps I will create a PR adding the symlink to the image.

**Comment by filbranden** [comment]
I just tried that, it didn't seem to fix the issue.
https://github.com/filbranden/vimgolf/runs/8129651541?check_suite_focus=true
Let me poke around some more.

**Comment by miketimofeev** [comment]
Intrestingly, the libs are in the `usr/local/opt/postgresql/lib/postgresql@14` not in the `usr/local/opt/postgresql/lib/`

**Comment by miketimofeev** [comment]
there is a discussion around it already https://github.com/orgs/Homebrew/discussions/3579

**Comment by filbranden** [comment]
Yes! This worked:

```
ln -s postgresql@14 /usr/local/opt/postgresql
ln -s postgresql@14/libpq.5.dylib /usr/local/opt/postgresql/lib/libpq.5.dylib
```

See commit https://github.com/filbranden/vimgolf/commit/c7cfad62f65261f1164b3d8c35ac845fe5298ec0

And corresponding run https://github.com/filbranden/vimgolf/runs/8130217463?check_suite_focus=true

Nice that you found a discussion about it already, thanks for that!

If you'd like to push a workaround for the issue in the meantime, I'd appreciate it.

**Comment by miketimofeev** [comment]
I would wait for a bit for the next steps from Homebrew before implementing a workaround. Would you mind putting your two cents in that discussion?

**Comment by filbranden** [comment]
Consider shipping `brew install libpq` in the MacOS image, it seems that specific Homebrew formula is more appropriate for users of the library instead of the version shipped with the database server itself.

**Comment by VadVergasov** [comment]
This issue with PostgreSQL also affects CMake and CMake couldn't find PostgreSQL on machine.

**Comment by miketimofeev** [comment]
@VadVergasov would you mind participating in the homebrew discussion as I'm afraid we can't control the way homebrew installs postgres? https://github.com/orgs/Homebrew/discussions/3579

**Comment by filbranden** [comment]
@miketimofeev It turns out the issue I was having was due to the bundle cache of the `ruby/setup-ruby` action. It had cached the gems built in the previous image, with the previous directory layout of the Postgres homebrew install, but then it was trying to use the same binaries on the new layout.

In debugging the issue, I noticed that building from source fixed the problem correctly, so then I found more about the bundle cache and how to [force a rebuild](https://github.com/ruby/setup-ruby#dealing-with-a-corrupted-cache) which is what is needed here.

I solved that with https://github.com/filbranden/vimgolf/commit/fa2a5a80e4f6f88e6343c2caca2e7cc3815567b3 and running it through GitHub Actions confirms [it fixed the issue](https://github.com/filbranden/vimgolf/runs/8141356065?check_suite_focus=true), so I'm pushing this upstream in PR igrigorik/vimgolf#360.

Thanks for your help!
